### PR TITLE
Companies component

### PR DESCRIPTION
**Pull request by jamespetran**
added backend companies stuff (router)
added companies redux stuff (saga and reducer)
added compatibility to companies component, using example array hardcoded in.

TODO still:
- actually make api call (using backend companies.router)
- formulate api call for specific company details
- formulate pagination for company
- make pagination component (if necessary - only render when more than x results exist etc)
- create rules for scores in back end
- add rules grabbing module 
- do....stuff with the rules based on company metrics

### CODE REVIEW

**Comment by jamespetran** [comment]
- please review the PR and add comments if you see anything that looks out of place. Happy to push more commits to make it all work better **before** we merge with main...don't approve it unless you're 100% happy with it

**Review by yemluck — APPROVED** [review]
Went through it and it looks GTM

**Pull request merged**
